scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/internal/propertyAccessors.scala

```diff
@@ -284,4 +356,68 @@ object propertyAccessors {
 
   }
 
+  sealed trait MethodAccessChecker {
+
+    def checkAccessIfMethodFound(targetClass: Class[_], onlyStaticMethods: Boolean = false)(
+        methodOpt: Option[Method]
+    ): Option[Method] = {
+      methodOpt.foreach(method => checkAccessForMethodName(targetClass, method.getName, onlyStaticMethods))
+      methodOpt
+    }
+
+    def checkAccessForMethodName(
+        targetClass: Class[_],
+        methodName: String,
+        onlyStaticMethods: Boolean
+    ): Unit = {
+      if (!methodAllowed(targetClass, methodName, onlyStaticMethods)) {
+        throw new IllegalStateException(
+          s"The $methodName method call on type ${targetClass.getSimpleName} is not allowed"
+        )
+      }
+    }
+
+    protected def methodAllowed(
+        targetClass: Class[_],
+        methodName: String,
+        onlyStaticMethods: Boolean
+    ): Boolean
+
+  }
+
+  object MethodAccessChecker {
+
+    def create(classDefinitionSet: ClassDefinitionSet, dynamicPropertyAccessAllowed: Boolean): MethodAccessChecker = {
+      if (!dynamicPropertyAccessAllowed) {
+        AllowedOnlyDefinedParameterlessMethods(classDefinitionSet)
+      } else {
+        AllMethodsAllowed
+      }
+    }
+
+  }
+
+  object AllMethodsAllowed extends MethodAccessChecker {
+
+    override protected def methodAllowed(
+        targetClass: Class[_],
+        methodName: String,
+        onlyStaticMethods: Boolean
+    ): Boolean = true
+
+  }
+
+  final case class AllowedOnlyDefinedParameterlessMethods(classDefinitionSet: ClassDefinitionSet)
```

Comment:
I think that more grammatically correct will be `OnlyDefinedParameterlessMethodsAllowed`. WDYT?